dissect/target/tools/shell.py

```diff
@@ -545,9 +561,20 @@ def cmd_stat(self, args, stdout):
         if not path:
             return
 
-        s = path.stat() if args.dereference else path.lstat()
+        symlink = f"-> {path.readlink()}" if path.is_symlink() else ""
+
+        if path.is_symlink() and args.dereference:
+            s = path.stat()
+        elif path.is_symlink():
+            s = path.lstat()
+        elif args.dereference:
+            s = path.stat()
+        else:
+            s = path.stat()
```

Comment:
This looks very duplicate, shouldn't it still be `s = path.stat() if args.dereference else path.lstat()`?